tools/loader/ImageClassifier.cpp

```diff
@@ -100,11 +114,30 @@ void loadImagesAndPreprocess(const llvm::cl::list<std::string> &filenames,
   inputImageData->reset(ElemKind::FloatTy, batchDims);
   auto IIDH = inputImageData->getHandle<>();
 
+  // assign mean and stddev for input normalization
+  llvm::ArrayRef<float> mean;
+  llvm::ArrayRef<float> stddev;
+  if (!meanValues.empty()) {
+    GLOW_ASSERT(meanValues.size() == numChannels &&
+                "Number of mean values != input channels");
+    mean = meanValues;
+  } else {
+    mean = std::vector<float>(numChannels, 0.f);
```

Comment:
This is going to cause a use after free. The vector here is constructed locally and will be destroyed after we leave the else scope, leaving the `mean` ArrayRef pointing to dealloc'd memory.

You can instead just reuse `meanValues`/`stddevValues` -- I'm not sure if you can set them directly (e.g. `meanValues = std::vector<float>(numChannels, 0.f);`), but at least `llvm::cl::list` does have a `addValue()` that you can use to append either the default values, or `imagenetNormMean`/`imagenetNormStd` if `useImagenetNormalization`.

Reply:
Now using `zeroMean` instead of `std::vector<float>(numChannels, 0.f);`.